locks/locks.go

```diff
@@ -1,22 +1,49 @@
 package locks
 
-var lockRequests chan *lockRequest
+var lockRequests = make(chan *lockRequest)
 
 func init() {
-	lockRequests = make(chan *lockRequest)
 	go locker()
 }
 
+type Locker interface {
+	Lock() Unlocker
+}
+
 // Unlocker Interface for unlocking a key that was provided to Lock()
 type Unlocker interface {
 	Unlock()
 }
 
+type KeyLockerImpl struct {
+	key interface{}
+}
+
+type NopLockerImpl struct{}
+
+var theNopLocker = &NopLockerImpl{}
+
+func NopLocker() *NopLockerImpl {
+	return theNopLocker
+}
+
+func (nl *NopLockerImpl) Lock() Unlocker {
+	return nl
+}
+
+func (nl *NopLockerImpl) Unlock() {
+	return
+}
+
+func KeyLocker(key interface{}) *KeyLockerImpl {
+	return &KeyLockerImpl{key: key}
+}
+
 // Lock Provides an application-wide lock on the provided key.
 // If the lock is obtained, it will return an Unlocker.
 // If the lock is not successfully obtained, nil will be returned.
-func Lock(key string) Unlocker {
```

Comment:
The Lock api is used by other projects on its own. From either a design or impl perspective, do you see any harm in keeping a backwards compatibel version of this API in addition to what you've added?